h/util/auth_client.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+"""Some shared utility functions for manipulating auth_client data."""
+from __future__ import unicode_literals
+import re
+
+
+def split_client(clientid):
+    """Return the ID and authority parts from the given clientid as a dict.
+
+    For example if userid is u'client:04465aaa-8f73-11e8-91ca-8ba11742b240@hypothes.is' then return
+    {'id': u'04465aaa-8f73-11e8-91ca-8ba11742b240', 'authority': u'hypothes.is'}'
+
+    :py:attr:`~h.models.user.AuthClient.id` must be a valid Python UUID
+
+    :raises ValueError: if the given clientid isn't a valid clientid or if the
+                        supplied AuthClient.id isn't a valid UUID
+
+    """
+    # Validates substring is a valid python UUID
+    uuid_match = r'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
+    match = re.match(r'^client:(' + uuid_match + r')@(.*)$', clientid)
```

Comment:
I opted to collapse the regex matching into a single go here, versus treating a match fail on UUID as a separate match—result being that both an invalid overall format and an invalid UUID (within an otherwise valid format) will raise the same ValueError—let me know if this is no good.